resolvers/dotnet/Azure.DigitalTwins.Validator/ValidateDTMI.cs

```
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Azure.DigitalTwins.Validator.Exceptions;
+
+
+namespace Azure.DigitalTwins.Validator
+{
+    public static partial class Validations
+    {
+
+        public async static Task<bool> ValidateDTMI(this FileInfo fileInfo)
+        {
+            var fileText = await File.ReadAllTextAsync(fileInfo.FullName);
+            return ValidateDTMI(fileText, fileInfo.FullName);
+        }
+        public static bool ValidateDTMI(string fileText, string fileName = "")
+        {
+            var dtmiRegex = new Regex("^dtmi:(?:_+[A-Za-z0-9]|[A-Za-z])(?:[A-Za-z0-9_]*[A-Za-z0-9])?(?::(?:_+[A-Za-z0-9]|[A-Za-z])(?:[A-Za-z0-9_]*[A-Za-z0-9])?)*;[1-9][0-9]{0,8}$");
+            var versionRegex = new Regex(";[1-9][0-9]{0,8}$");
+
+            var model = JsonDocument.Parse(fileText).RootElement;
+            JsonElement rootId;
+            if(model.TryGetProperty("@id", out rootId)){
```

nit, you could flip the logic here for less branching like 
`if (!model.TryGetProperty("@id", out rootId)){ throw new MissingDTMIException(fileName); }`